TUM Campus App/StudyRoomsTableViewController.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  StudyRoomsTableViewController.swift
+//  TUM Campus App
+//
+//  Created by Max Muth on 18.12.16.
+//  Copyright © 2016 LS1 TUM. All rights reserved.
+//
+
+import UIKit
+import AYSlidingPickerView
+
+class StudyRoomsTableViewController: UITableViewController, DetailView {
+    
+    var delegate: DetailViewDelegate?
+    
+    var pickerView = AYSlidingPickerView()
+    var barItem: UIBarButtonItem?
+    
+    var refresh = UIRefreshControl()
+    func refresh(_ sender: AnyObject?) {
+        roomGroups.removeAll()
+        studyRooms.removeAll()
+        delegate?.dataManager().getAllStudyRooms(self)
+    }
+
+    
+    var roomGroups = [StudyRoomGroup]()
+    var studyRooms = [StudyRoom]() // Rooms of all available groups
+    var currentGroup: StudyRoomGroup? {
+        didSet {
+            title = currentGroup?.name
+            currentRooms = self.studyRooms
+                .filter() { (self.currentGroup?.roomNumbers.contains($0.roomNumber)) ?? false }
+                .sorted() { $0.status.sortIndex < $1.status.sortIndex }
+        }
+    }
+    var currentRooms = [StudyRoom]() { // Rooms for the group that is currently selected
+        didSet {
+            tableView.reloadData()
+            refresh.endRefreshing()
+        }
+    }
+    
+}
+
+extension StudyRoomsTableViewController: TumDataReceiver {
+    
+    func receiveData(_ data: [DataElement]) { // Is called twice, once with [StudyRoomGroups] as argument, once for [StudyRooms]
+        for item in data {
```

Comment:
Since we're already hunting down uncessesary loops:
```
roomGroups.append(contentsOf: data.flatMap { $0 as? StudyRoomGroup })
studyRooms.append(contentsOf: data.flatMap { $0 as? StudyRoom })
```
should do it ;)